### PR TITLE
feat: Clone Repo Dialog

### PR DESCRIPTION
**Pull request by shaurya-harness**
adds Clone Repo Dialog

Screenshot 2024-12-18 at 6 27 51 PM
Screenshot 2024-12-18 at 6 28 02 PM

### CODE REVIEW

File: apps/gitness/src/pages-v2/repo/repo-summary.tsx [hunk 1]
```diff
@@ -122,6 +123,11 @@ export default function RepoSummaryPage() {
   const [createdTokenData, setCreatedTokenData] = useState<(TokenFormType & { token: string }) | null>(null)
   const [successTokenDialog, setSuccessTokenDialog] = useState(false)
 
+  const onManageToken = () => {
+    // TODO: update when settings for tokens page is added
+    navigate('/settings')
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages-v2/repo/repo-summary.tsx:128]: can this be a `<Link>` instead?

File: packages/ui/src/components/input.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/components/input.tsx:1]: this should be reviewed by XD team I think

File: packages/ui/src/views/repo/components/token-dialog/clone-credential-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  const { t } = useTranslationStore()
+  const { setValue } = useForm<TCloneCredentialsDialog>({
+    resolver: zodResolver(formSchema),
+    defaultValues: tokenData
+  })
+  useEffect(() => {
+    if (tokenData) {
+      setValue('identifier', tokenData.identifier)
+      setValue('lifetime', tokenData.lifetime)
+      setValue('token', tokenData.token)
```
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/repo/components/token-dialog/clone-credential-dialog.tsx:54]: why is this needed when `tokenData` is already being passed to `defaultValues` above?

**Review by abhinavrastogi-harness — APPROVED** [review]

File: packages/ui/src/views/repo/components/token-dialog/clone-credential-dialog.tsx (hunk 2 above)
Comment (shaurya-harness) [reply packages/ui/src/views/repo/components/token-dialog/clone-credential-dialog.tsx:54]: yeah, not needed, will remove this

File: packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button>
+          <Icon name="clone" />
+          &nbsp; {t('views:repos.clone', 'Clone')}
+        </Button>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent className="w-[328px] p-0" align="end">
+        <div className="px-3 pt-2">
```
Comment (andrewgolovanov) [inline packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx:48]:
```suggestion
        <div className="px-4 pt-4">
```
It seems that the Tabs component also needs updated padding—currently, it’s set to `px-3` by default, but it should be `px-4`

image

File: packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        </Tabs>
+        <div className="px-5 py-4">
+          {currentTab === 'https' ? (
+            <>
+              <Input
+                id="httpsUrl"
+                readOnly
+                value={httpsUrl}
+                variant="extended"
+                className="text-tertiary-background mb-2"
```
Comment (andrewgolovanov) [inline packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx:94]:
```suggestion
                className="text-foreground-2"
```

image

File: packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+                right={<CopyButton name={httpsUrl} />}
+              />
+              <div className="flex items-center my-2">
```
Comment (andrewgolovanov) [inline packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx:97]:
```suggestion
              <div className="flex items-center mt-4">
```
image

File: packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                <span className="ml-1 text-tertiary-background">
```
Comment (andrewgolovanov) [inline packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx:98]:
```suggestion
                <span className="text-foreground-4">
```

We can also remove the left margin.

image

File: packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+                  {t('views:repos.generateCredential', 'Please generate a clone credential if its your first time.')}
+                </span>
+              </div>
+              <div className="flex items-center mb-2">
```
Comment (andrewgolovanov) [inline packages/ui/src/views/repo/repo-summary/components/clone-repo-dialog.tsx:102]:
```suggestion
              <div className="flex items-center mt-4">
```

**Review by andrewgolovanov — APPROVED** [review]
